src/main/java/hudson/tasks/Mailer.java

```diff
@@ -627,11 +627,11 @@ public FormValidation doCheckDefaultSuffix(@QueryParameter String value) {
         /**
          * Send an email to the admin address
          * @throws IOException in case the active jenkins instance cannot be retrieved
-         * @param smtpServer name of the SMTP server to use for mail sending
+         * @param smtpHost name of the SMTP server to use for mail sending
          * @param adminAddress Jenkins administrator mail address
-         * @param useSmtpAuth if set to {@code true} SMTP is used without authentication (username and password)
-         * @param smtpAuthUserName plaintext username for SMTP authentication
-         * @param smtpAuthPasswordSecret plaintext password for SMTP authentication
+         * @param authentication if set to {@code true} SMTP is used without authentication (username and password)
```

Comment:
Mmmmm this name is a bit confusing. Can't we use something like useAuthentication?

Reply:
As far as I understand the code, the name of the method paramter must be the same as the one used in the HTML/Jelly form. This is the property used in the JSON request when the sendTestMail button is clicked in the UI.

see: https://github.com/jenkinsci/mailer-plugin/blob/fafdd53cb339c2aea49811e36a7af1846df7e8cd/src/main/resources/hudson/tasks/Mailer/global.jelly#L36
and
https://github.com/jenkinsci/mailer-plugin/blob/fafdd53cb339c2aea49811e36a7af1846df7e8cd/src/main/resources/hudson/tasks/Mailer/global.jelly#L55

Should I rename the `authentication` field (introduced/changed in #61)?

Reply:
I'm not sure, in [previous versions](https://github.com/jenkinsci/mailer-plugin/blob/mailer-1.24/src/main/resources/hudson/tasks/Mailer/global.jelly#L35) the field was created in the Jelly file and the type was different?

But if this works I'm not requesting a change, just trying to understand.

Reply:
I'm not so familiar with jelly, but in the previous version this was an `<f:optionalBlock` now it is an `<f:optionalProperty`.
This optional property is based on the [SMTPAuthentication.java](https://github.com/jenkinsci/mailer-plugin/blob/master/src/main/java/hudson/tasks/SMTPAuthentication.java) class and thus generating a `username` and `password` field.
The property is created here and then bound to the form https://github.com/jenkinsci/mailer-plugin/blob/eb5c215b0e7a9a6fcd581285eb895ec4a486db2b/src/main/java/hudson/tasks/Mailer.java#L271

As far as my testing goes, this change works.

Reply:
Thanks for the explanations!